### ISSUE TITLE
Questionable IDE0067 when awaiting class-member Task containing IDisposable

### ISSUE THREAD

**Issue by Tragetaschen** [dotnet/roslyn#36643]
**Version Used**: 
16.2.0 Preview 2.0

**Steps to Reproduce**:

```csharp
using System.IO;
using System.Threading.Tasks;

public class C
{
    private readonly Task<FileStream> fileStreamTask;
    public C() => fileStreamTask = Task.Run(() => File.OpenRead("")); // async resource initialization

    public async Task M()
    {
        var stream = await fileStreamTask; // IDE0067
        await stream.FlushAsync(); // more code using `stream`
    }
    // Dispose[Async]
}
```

**Expected Behavior**:
The Task holding the `IDisposable` has class scope. Disposing the contents breaks successive calls to M(). IDE0067 should not be reported.

**Actual Behavior**:
IDE0067 and a code fix that breaks the semantics.

On a side note: The cursor positions that show the code fix doesn't match the indicator: The smudge is placed under the letters `aw` of `await`, while the code fix is only available before and after the `=` and before the `a`.

**Comment by sharwell** [comment]
Agree. A task only transfers ownership of its result when it goes out of scope afterwards.

**Comment by mavasani** [comment]
Re-opening as the fix has not yet flowed into Roslyn.
